quantecon/_lqnash.py

```diff
@@ -118,28 +118,27 @@ def nnash(A, B1, B2, R1, R2, Q1, Q2, S1, S2, W1, W2, M1, M2,
         F10 = F1
         F20 = F2
 
-        G2 = solve(np.dot(B2.T, P2.dot(B2))+Q2, v2)
-        G1 = solve(np.dot(B1.T, P1.dot(B1))+Q1, v1)
-        H2 = np.dot(G2, B2.T.dot(P2))
-        H1 = np.dot(G1, B1.T.dot(P1))
+        G2 = solve(B2.T @ P2 @ B2 + Q2, v2)
+        G1 = solve(B1.T @ P1 @ B1 + Q1, v1)
+        H2 = G2 @ B2.T @ P2
+        H1 = G1 @ B1.T @ P1
 
         # break up the computation of F1, F2
-        F1_left = v1 - np.dot(H1.dot(B2)+G1.dot(M1.T),
-                           H2.dot(B1)+G2.dot(M2.T))
-        F1_right = H1.dot(A)+G1.dot(W1.T) - np.dot(H1.dot(B2)+G1.dot(M1.T),
-                                                H2.dot(A)+G2.dot(W2.T))
+        F1_left = v1 - (H1 @ B2 + G1.dot(M1.T)) @ (H2 @ B1 + G2.dot(M2.T))
+        F1_right = H1 @ A + G1.dot(W1.T) - (H1 @ B2 + G1.dot(M1.T)) @ \
+            (H2 @ A + G2.dot(W2.T))
```

Comment:
Here are some `.dot`. Are these kept intentionally?

Reply:
> I left only a few np.dot in cases involving scalars. I noticed during the process that [the documentation](https://quanteconpy.readthedocs.io/en/latest/tools/lqnash.html) for parameter A of nnash function in _lqnash.py seems to indicate that A can be scalar. However, based on the setup of the formula, it seems to me that A is unlikely to be a scalar, given the two-player setup here. Please kindly correct me if I am wrong.

Many thanks @Smit-create. Yes, as noted before, these are cases where scalars are allowed, and `@` is not capable of multiplying matrices with scalars.